src/handler/mod.rs

```diff
@@ -758,8 +761,11 @@ impl Handler {
                 }
             }
         }
-        self.new_session::<P>(node_address, session, Some(auth_message_nonce))
+        self.new_session::<P>(node_address.clone(), session, Some(auth_message_nonce))
             .await;
+        // We could have pending messages that were awaiting this session to be
+        // established. If so process them.
+        self.send_pending_requests::<P>(&node_address).await;
```

Comment:
This appears to be duplicated from line 850-852, the only other time we use `new_session()`. Is it worth moving this code into `new_session()`?

Reply:
Looks good, thank you!